### DISCUSSION TITLE
Use dashboard input to alter scheduled time.

### DISCUSSION THREAD

**Discussion by deek042** [jensrossbach/node-red-contrib-chronos#79] [Q&A] [answered]
I'm still rather fairy new to Node-red.
I was wondering what I would as msg.payload(or whatever) to set the value of time in a scheduler node.
I'm trying to set the time with a input box on a dashboard.

**Comment by deek042** [comment]
Currently trying to figure out the Dynamic Trigger as stated in the wiki.

**Comment by jensrossbach** [comment, marked as answer]
Hi @deek042, there are several possibilities. One way that I went already in the past was to setup a text input widget configured as time picker in a Node-RED dashboard. The output of such widget is the time in milliseconds since midnight. This time can be written to a context variable in the following JSON format:

``` javascript
{
    "type":   "time",
    "value":  time_value_from_widget,
    "offset": 0,
    "random": false
}
```

Then you can configure the scheduler node to use that context variable as trigger for an event and it will fire as soon as the time has been reached. You can additionally use a switch widget on the dashboard to turn the scheduler node on or off (by sending a message with payload equal to true or false to the node when the state of the switch changes). You send a message with payload equal to "reload" to the scheduler node each time you have written an updated time value to the context variable. The text input time picker widget can be configured to only send an output when the focus left.

Here is a very simple flow that should do the above:
![image](https://user-images.githubusercontent.com/5118924/151037316-a224c743-482a-4670-b645-5198b4f809e2.png)

This is the flow:
``` json
[{"id":"24a2b58d3fcbf32f","type":"tab","label":"Scheduler","disabled":false,"info":"","env":[]},{"id":"fadd53876e1d5854","type":"ui_text_input","z":"24a2b58d3fcbf32f","name":"","label":"","tooltip":"","group":"2abe341a.254e8c","order":1,"width":0,"height":0,"passthru":true,"mode":"time","delay":300,"topic":"topic","sendOnBlur":true,"className":"","topicType":"msg","x":600,"y":360,"wires":[["7451ace041a92be0"]]},{"id":"7451ace041a92be0","type":"change","z":"24a2b58d3fcbf32f","name":"","rules":[{"t":"set","p":"time","pt":"flow","to":"{\t   \"type\": \"time\",\t   \"value\": payload,\t   \"offset\": 0,\t   \"random\": false\t}\t","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":770,"y":360,"wires":[["7c1f13683ce34815"]]},{"id":"7c1f13683ce34815","type":"change","z":"24a2b58d3fcbf32f","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"reload","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":960,"y":360,"wires":[["0850e3cc5766e9f5"]]},{"id":"0850e3cc5766e9f5","type":"chronos-scheduler","z":"24a2b58d3fcbf32f","name":"","config":"5e581578.25c88c","schedule":[{"trigger":{"type":"flow","value":"time"},"output":{"type":"msg","property":{"name":"payload","type":"bool","value":"true"}}}],"multiPort":false,"disabled":false,"outputs":1,"x":1140,"y":360,"wires":[["9692e896d21dda9e"]]},{"id":"9692e896d21dda9e","type":"debug","z":"24a2b58d3fcbf32f","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1310,"y":360,"wires":[]},{"id":"2abe341a.254e8c","type":"ui_group","name":"Test","tab":"6ef3ea88.2f9554","order":1,"disp":true,"width":"6","collapse":false},{"id":"5e581578.25c88c","type":"chronos-config","name":"Berlin","sunPositions":[{"angle":-4,"riseName":"blueHourDawn","setName":"blueHourDusk"},{"angle":-8,"riseName":"blueHourDawnBegin","setName":"blueHourDuskEnd"}]},{"id":"6ef3ea88.2f9554","type":"ui_tab","name":"Test","icon":"dashboard","order":1,"disabled":false,"hidden":false}]
```

**Reply by jensrossbach** [reply]
One thing to mention: the above flow is not 100% perfect and complete. For instance it has the problem that it doesn't work right after Node-RED has started as there is no context variable available yet (you need to set a time in the input text input widget and then restart the flow to have it working). So some more stuff needs to be done in order to get it more robust.

**Reply by deek042** [reply]
Thank you!!! Before your response, I had it working by injecting one of the wiki examples for dynamic input.  But then I was stumbling on how to get the text input to work.
